src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.cs

```diff
@@ -36,5 +32,18 @@ protected DirectoryInfo CreateSelfReferencingSymbolicLink()
         protected string GetRandomDirPath()  => Path.Join(ActualTestDirectory.Value, GetRandomDirName());
 
         private Lazy<string> ActualTestDirectory => new Lazy<string>(() => GetTestDirectoryActualCasing());
+
+        /// <summary>
+        /// Changes the current working directory path to a new temporary directory.
+        /// Important: Make sure to call this inside a remote executor to avoid changing the cwd for all tests in same process.
```

Comment:
it would have to be something like this
`Debug.Assert(Environment.StackTrace.Contains("RemoteExecutor"));`
or it would be reasonable to add a static flag in RemoteExecutor so you could just test RemoteExecutor.IsInRemoteExecutor.